tools/run_tests/run_tests.py

```diff
@@ -712,7 +713,7 @@ def test_specs(self):
         runtime_cmd = ['mono']
 
     specs = []
-    for assembly in tests_by_assembly.iterkeys():
```

Comment:
We use `six` for all 2-and-3-together compatibility requirements, so `six.iterkeys(tests_by_assembly)` in that case?